### DISCUSSION TITLE
Web UI Feedback

### DISCUSSION THREAD

**Discussion by jugeeya** [jugeeya/UltimateTrainingModpack#199] [General]
Please report feedback on the Web UI menu here. It is in the latest beta, it is activated with Special + UpTaunt.

**Comment by tehxeno** [comment]
### WebUI critique:
* Opening the menu has a short ~1 second load that could feel disruptive if going into the menu to change options frequently.
* I don't like that I can't "loop" the scrolling - so pressing up at the very top of the list to move to the bottom.
* I DO like that the menu works with the touch screen, although I don't think I'd do my training in mobile mode all that often.
* I think "Mash in neutral" should be a menu options right below the Mash/Follow-up toggles, like it is in the Tesla menu
* There is a lot of white space in each top level menu option - I would either use the white space to list sub-options selected, or split the menu into two columns like https://cdn.discordapp.com/attachments/407970595423125506/832356168508768286/unknown.png
     * If listing sub-options, it could be something like
    Mash Toggles - Air_Dodge, Jump, Nair...
* It might be nice to have the sections like the Tesla menu has and use them like accordions to pivot the sub-options and be able to collapse sections you don't use often to reduce scrolling
* Ledge Delay - if we didn't have a slider, it would be nice to have a "select all" or "deselect all" like the Tesla All/None options
     * These would be nice to have back at any section that has multi-select options. I noticed that this is a thing for ledge and defensive options as well

### WebUI bugs:
* (bug? Probably not intended use) The WebUI loads the currently active setting changes - however the Tesla menu does NOT load the active settings from WebUI and will overwrite the options with the Tesla menu selections once it is closed.
* There's a mysterious blank option under Shield Tilt, DI, SDI, and airdodge direction - I assume it's supposed to be "neutral" or no directional input.
* Input delay does not have any options - the only option is displays the selected value from the Tesla menu
* Mash in Neutral does not work when toggled on from the WebUI, but does work from Tesla
* There's no option for "save states" to get the help text about how to use them
* There's no option for "Save damage" like there is in the Tesla menu
* Inputting UpTaunt, then B does not open the menu, you must enter B then UpTaunt
* Holding B for a while, then tapping UpTaunt will buffer opening the menu a second time (it opens immediately after closing the menu.) closing the second menu open will close as expected.
* Inputting UpTaunt + B like a smash attack opens the menu three times. A new menu opens immediately after closing the menu.

**Comment by Apep-the-snek** [comment]
> Please report feedback on the Web UI menu here. It is in the latest beta, it is activated with Special + UpTaunt.

just bring it back lol, i already requested it in issues in more detail.
